cli/config/credentials/file_store.go

```diff
@@ -8,27 +8,28 @@ import (
 type store interface {
 	Save() error
 	GetAuthConfigs() map[string]types.AuthConfig
+	GetFilename() string
 }
 
-// fileStore implements a credentials store using
+// FileStore implements a credentials store using
 // the docker configuration file to keep the credentials in plain text.
-type fileStore struct {
+type FileStore struct {
```

Comment:
Instead of exporting this (something we want to deprecate) how about adding a new `IsNotSecure()` method to `fileStore`, and creating the interface in `cli/command/registry/login.go` to check it?

```go

type notSecure interface {
    IsNotSecure() bool
}

if _, isNotSecure := creds.(notSecure); isNotSecure {
 ...
}

```

Reply:
I thought about this, but as it stands there are two backends: fileStore and nativeStore, which is what implements the docker credential helpers interface. You could imagine a credential helper that just wrote plain text to a file, which would not be secure, but writing "!isNotSecure" seems to imply that the backend is secure, when it's not: nativeStore can't vouch for what's on the other end.

Reply:
Ok, then how about `IsDeprecated()` and `type Deprecated interface{...}` ?

Reply:
I thought about this too, but in the event that we don't deprecate this based on discussion, I think we should still merge this patch, but tagging it as deprecated would be incorrect. We could cast by GetFilename, since that is unique to the FileStore, but it doesn't really describe what we're trying to detect. So I dunno :)

Reply:
Feel free to pick some other interface, my concern is just that it doesn't make sense to export this entire struct if all we need is some way detect that it is of this type.

Another option is to export an `IsFileStore()`. That way we're only exporting one function and not  the entire struct.